frontend/src/app/shopping-cart-list/shopping-cart-list.component.ts

```diff
@@ -11,43 +10,50 @@ import { Subject } from 'rxjs/Subject';
 })
 export class ShoppingCartListComponent implements OnChanges {
 
+  private products : Map<number, ProductModel> = new Map<number, ProductModel>();
+
   @Input("items")
   private cartList : Array<ShoppingCartItem> = [];
-  private products : Map<number, ProductModel> = new Map<number, ProductModel>();
-  
+
   @Output("itemChange")
   private itemQtyEmitter = new EventEmitter<ShoppingCartItem>();
 
   @Output("itemDelete")
-  private itemDelEmitter = new EventEmitter<ShoppingCartItem>(); 
-
+  private itemDelEmitter = new EventEmitter<ShoppingCartItem>();
 
   constructor() { }
 
   ngOnChanges() {
-    for(let item of this.cartList){
-      if(!(item.productId in this.products)){
-        item.product.subscribe((product : ProductModel) => {
-          this.products[item.productId] = product;
-        })
-      }
-    }
+    this.fillProducts()
+  }
+
+  private fillProducts():void {
```

Comment:
bit inconsistent with `: void` it hasn't really been used elsewhere in the application.